src/lib/y2network/connection_config/base.rb

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/interface"
+
+module Y2Network
+  module ConnectionConfig
+    # This class is reponsible of a connection configuration
+    class Base
+      # @return [Interface, nil]
+      attr_accessor :interface
+      # @return [String] Bootproto (static, dhcp, none)
```

Comment:
ibft is also possible value. And what about zeroconf ( aka autoip ) and autoip combination with dhcp and also dhcp version specific ( dhcp4 really helps with speed on ipv4 only network ).

Reply:
Sure, I could put all the options

static*|dhcp|dhcp4|dhcp6|autoip|dhcp+autoip|auto6|6to4|none

Reply:
or point to possible values to enum, which is ideal solution as then we can have storage like `to_human_string` method, that returns human friendly name.